File: Python/0064. Lucky Number.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
            n,x,y,z = map(int,input().strip().split())
            
            def gcd(a,b):
                while a%b:
                    a = a%b
                    if a<b:
                        a,b = b,a
                return b
            
            def lcm(a,b):
                return a*b//gcd(a,b)
            
            print(n//x+n//y+n//z-n//lcm(x,y)-n//lcm(x,z)-n//lcm(y,z)+n//lcm(x,lcm(y,z)))
            
        except EOFError:
            break
    
if __name__ == "__main__":
    func()

File: C++/0064. Lucky Number.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.
int gcd(int a, int b){
    while(a%b!=0){
        a = a%b;
        if(a<b) swap(a,b);
    }
    return b;
}

int lcm(int a,int b){
    return a*b/gcd(a,b);
}

int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 

    int n,x,y,z;
    while(cin>>n>>x>>y>>z){
        cout<<n/x+n/y+n/z-n/lcm(x,y)-n/lcm(x,z)-n/lcm(y,z)+n/lcm(x,lcm(y,z))<<endl;       
    }
    return 0;
}
